pkg/ingester/stream.go

```diff
@@ -240,7 +240,7 @@ func (s *stream) Push(
 		// Check if this this should be rate limited.
 		now := time.Now()
 		if !s.limiter.AllowN(now, len(entries[i].Line)) {
-			failedEntriesWithError = append(failedEntriesWithError, entryWithError{&entries[i], ErrStreamRateLimit})
+			failedEntriesWithError = append(failedEntriesWithError, entryWithError{&entries[i], &validation.ErrStreamRateLimit{RateLimit: flagext.ByteSize(s.limiter.lim.Limit()), Labels: s.labelsString, Bytes: flagext.ByteSize(len(entries[i].Line))}})
```

Comment:
Can we cache `s.limiter.lim.Limit()` so we aren't calling it for every rate limited entry?